### ISSUE TITLE
Add utility to query pubs and relationships by name

### ISSUE THREAD

**Issue by gabestein** [pubpub/platform#737]
## Motivation

- So developers can ergonomically build the ability to search for pubs by name in, e.g., the editor or in a relationship field input component

## Requirements

- Ability to search all pubs in a community by a string
- Ability to filter search by:
  - type of pub (e.g., only return pubs that are of type "person")
  - filter only to certain relationship fields (e.g. give me all related pubs that are related via the "author" relation field on this pub)
  - pubs which have a relationship to a specified pub or set of pubs (e.g. give me all the related pubs to the "My journal article" pub)

Requires #735

**Comment by gabestein** [comment]
For Monday: let's try to break down, particularly for the editor, all the different "search by title" functions this query is fulfilling, and if needed, break this into separate query functions.

**Comment by tefkah** [comment]
@gabestein yes, this was sort of already possible with the initial FTS behavior, but is now integrated with the normal querying stuff thanks to #1197
